### PR TITLE
feat(payment): PAYPAL-2758 banner in Paywall 

### PR DESCRIPTION
**Pull request by bc-nick**
## What?

`PayLater Message` in Paywall section on Checkout page

## Why?

To display `PayLater Messages` in the appropriate place in the Paywall section (Payment step)

Depends on: https://github.com/bigcommerce/checkout-sdk-js/pull/2076

## Testing / Proof

Screenshot 2023-08-01 at 13 58 03

https://github.com/bigcommerce/checkout-js/assets/99336044/8e813594-6175-42f3-a993-dad827628cf7

@bigcommerce/checkout

### CODE REVIEW

**Comment by animesh1987** [comment]
Hey @bc-nick this seems like paypal related functionality, can we try and address it in the PaymentMethod or PaymentMethodTitle component?

File: packages/core/src/app/ui/form/ChecklistItem.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export interface ChecklistItemProps {
     htmlId?: string;
     isDisabled?: boolean;
     label: ReactNode | ((isSelected: boolean) => ReactNode);
+    description?: ReactNode | (() => ReactNode);
```
Comment (serhii-tkachenko) [inline packages/core/src/app/ui/form/ChecklistItem.tsx:17]: Is there any chance to pass a string to description?

Comment (bc-nick) [reply packages/core/src/app/ui/form/ChecklistItem.tsx:17]: Yes. We have this opportunity thanks to ReactNode

File: packages/core/src/app/payment/paymentMethod/PaymentMethodDescription.tsx
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodDescription.tsx:1]: The idea of `PaymentMethodDescription` might confuse developers because we want to keep `CheckoutListItem` simple, if a method requires complicated descriptions, then a custom component should be considered.
Just like Animesh suggested, could you please put this logic inside `PaymentMethodTitle` and treat it as a subtitle?

Comment (bc-nick) [reply packages/core/src/app/payment/paymentMethod/PaymentMethodDescription.tsx:1]: @bc-peng I have updated the PR to include your suggestion. Take a look please

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx [hunk 2]
```diff
@@ -285,33 +296,38 @@ const PaymentMethodTitle: FunctionComponent<
...
+                        selectedCardType={getSelectedCardType()}
+                    />
+                </div>
             </div>
-            <div className="paymentProviderHeader-cc">
-                <CreditCardIconList
-                    cardTypes={compact(method.supportedCards.map(mapFromPaymentMethodCardType))}
-                    selectedCardType={getSelectedCardType()}
-                />
+            <div className="paymentProviderHeader-subtitleContainer">
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx:327]: Could you please add a check here to only display this container whenever there is a subtitle?

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx [hunk 3]
```diff
@@ -52,7 +52,7 @@ describe('PaymentMethodTitle', () => {
                 ccNumber: '4111 1111 1111 1111',
                 paymentProviderRadio: getPaymentMethod().id,
             },
-            method: getPaymentMethod(),
+            method: getPaymentMethod()
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx:55]: Is this change necessary?

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx (hunk 2 above)
Comment (bc-nick) [reply packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx]: Sure, but I think it would be better if I remove it because the class name is not used at all right now. Thanks for the comment! Fixed

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx (hunk 3 above)
Comment (bc-nick) [reply packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx]: Fixed

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx [hunk 4]
```diff
@@ -661,4 +661,48 @@ describe('PaymentMethodTitle', () => {
...
+                <PaymentMethodTitleTest
+                    {...defaultProps}
+                    method={{
+                        ...defaultProps.method,
+                        id: PaymentMethodId.BraintreePaypalCredit,
+                    }}
+                />,
+            );
+
+            expect(screen.getByTestId('braintree-banner-container')).not.toBeUndefined();
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx:705]:
```suggestion
            expect(screen.getByTestId('braintree-banner-container')).toBeInTheDocument();
```

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx [hunk 5]
```diff
@@ -285,33 +296,36 @@ const PaymentMethodTitle: FunctionComponent<
...
+                </div>
+                <div className="paymentProviderHeader-cc">
+                    <CreditCardIconList
+                        cardTypes={compact(method.supportedCards.map(mapFromPaymentMethodCardType))}
+                        selectedCardType={getSelectedCardType()}
+                    />
+                </div>
             </div>
-        </div>
+            {subtitle instanceof Function ? subtitle({ onUnhandledError }) : subtitle}
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx:327]: I'm concerned that putting `subtitle` outside of `paymentProviderHeader-container` would cause merchants' existing CSS customizations to break. It also breaks the logic of the checkout's dom-structure.
Is it possbile to keep using `paymentProviderHeader-container` to hold everything?

Comment (bc-nick) [reply packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx]: Updated with using `flex-basis`

File: packages/core/src/scss/components/checkout/paymentProvider/_paymentProvider.scss [hunk 6]
```diff
@@ -5,17 +5,24 @@
...
     display: flex;
     justify-content: space-between;
+    flex-basis: 50%;
 }
 
 .paymentProviderHeader-img {
     align-self: center;
     display: flex;
     height: fontSize("large");
+    object-fit: contain;
```
Comment (bc-peng) [inline packages/core/src/scss/components/checkout/paymentProvider/_paymentProvider.scss:25]: Does the flex layout change the look of the header img?

File: packages/core/src/scss/components/checkout/paymentProvider/_paymentProvider.scss [hunk 7]
```diff
@@ -5,17 +5,24 @@
     align-items: start;
     display: flex;
     justify-content: space-between;
+    flex-wrap: wrap;
+}
+
+.paymentProviderHeader-subtitleContainer {
+    flex-basis: 100%;
```
Comment (bc-peng) [inline packages/core/src/scss/components/checkout/paymentProvider/_paymentProvider.scss:12]: This line forces `subtitleContainer` to a new row.
My concern is that we don't want to split `nameContainer` and `cc` evenly on the first row.
`space-between` is still our ideal layout.
Therefore, could you please remove `flex-basis: 50%;` in the PR ?

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx [hunk 8]
```diff
@@ -284,34 +295,45 @@ const PaymentMethodTitle: FunctionComponent<
...
-            >
-                {logoUrl && (
-                    <img
-                        alt={methodName}
-                        className="paymentProviderHeader-img"
-                        data-test="payment-method-logo"
-                        src={logoUrl}
-                    />
-                )}
+        <>
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx:307]:
```suggestion
        
```

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx [hunk 9]
```diff
@@ -284,34 +295,45 @@ const PaymentMethodTitle: FunctionComponent<
...
+                <div className="paymentProviderHeader-cc">
+                    <CreditCardIconList
+                        cardTypes={compact(method.supportedCards.map(mapFromPaymentMethodCardType))}
+                        selectedCardType={getSelectedCardType()}
+                    />
+                </div>
+                {getSubtitle()}
             </div>
-        </div>
+        </>
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx:336]:
```suggestion
        
```

File: packages/core/src/scss/components/checkout/paymentProvider/_paymentProvider.scss (hunk 6 above)
Comment (bc-nick) [reply packages/core/src/scss/components/checkout/paymentProvider/_paymentProvider.scss:25]: Yes

**Review by bc-peng — APPROVED** [review]
A tremendous amount of kudos to you. 👍

**Review by serhii-tkachenko — APPROVED** [review]
Nice work!
